### ISSUE TITLE
Handle coupled channels

### ISSUE THREAD

**Issue by danielballan** [NSLS-II/caproto#79]

**Comment by tacaswell** [comment]
Sorry for this being very rough, had this thought in the shower and wanted to get in down in any form before I forget (like I do many things these days (or @danielballan has been trolling me for a year now.... :smiling_imp: )).

```python

class IOCMeta(ActualMeta):
     
     def pvdb(self, format_data):
         return {chan.pv_format(**format_data): chan for chan is self.chanels()}

class MyIOC(IOCMeta):
     a = Single(data_class=FloatData, pv_format='{pr}:a')
     b = Mixed(data_class=FloatData, pv_format='{pr}:b')
     c = Mixed(data_class=FloatData, pv_format='{pr}:c')
     d = RO(data_class=StrData, pv_format='{pr}:d')

     def set_dbr_mixed(self, channel_that_was_set_from_outside, set_kwargs):
          # stuff
          yield from a.single_set_dbr(**set_kwargs)

```

We have a meta-class that lets us define IOC's declarativly (like ophyd), but in addition to the the counting fingers and toes, may change the class on the way by.  In the case of `Single` it does nothing, in the case of `Mixed` it changes `set_dbr` to delegate back to `set_dbr_mixed` on the parent and adds a `single_set_dbr` as the original `set_dbr`.  RO renames `set_dbr` -> `private_set_dbr` and does tells the server that it is read only.

To stand up a server then looks like

```python
await Context('0.0.0.0', find_next_tcp_port(), MyIOC().pvdb({'pr': 'PV'}).run()
```

**Comment by tacaswell** [comment]
It is also tempting to build a ophyd mirror with nested devices.

**Comment by danielballan** [comment]
The original title of this issue (Revisit and finalize API in _data.py.) has been provisionally achieved by #96. I have renamed it to "Handle coupled channels" to align it with Tom's comment, which is I think an interesting way to go.

**Comment by tacaswell** [comment]
A comment following up on my conversation with @klauer 

It is important that the API for updating a fields state (alarm, limits, value) and triggering monitors needs to be easily accessible from both outside (via exposing it to the server to set via opics write commands) and from the inside.

**Comment by klauer** [comment]
Largely addressed by #157; closing for now.
